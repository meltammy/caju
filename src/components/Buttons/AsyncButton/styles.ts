import styled from "styled-components";
import { Spinner } from "~/components/Spinner";

export const StyledSpinner = styled(Spinner)`
  width: 5px;
  height: 5px;
  border-width: 3px;
`;
