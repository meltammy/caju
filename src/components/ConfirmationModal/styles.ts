import styled from "styled-components";

export const ModalOverlay = styled.div<{ $isOpen: boolean }>`
  display: ${(props) => (props.$isOpen ? "flex" : "none")};
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.5);
  justify-content: center;
  align-items: center;
`;

export const ModalContainer = styled.div`
  background: white;
  border-radius: 8px;
  padding: 20px;
  min-width: 500px;
  max-width: 95vw;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
  z-index: 999;

  @media screen and (max-width: 1200px) {
    min-width: 50vw;
    margin: 1rem;
  }
`;

export const Title = styled.h2`
  margin: 0;
`;

export const Description = styled.p`
  margin: 10px 0 20px;
`;

export const ButtonContainer = styled.div`
  display: flex;
  gap: 1rem;

  @media screen and (max-width: 1200px) {
    flex-direction: column-reverse;
  }

  button {
    width: 100%;
    place-content: center;
  }
`;
