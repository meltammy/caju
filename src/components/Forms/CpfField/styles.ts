import ReactInputMask from "react-text-mask";
import styled from "styled-components";

export const Input = styled(ReactInputMask)`
  width: 100%;
  padding: 0 0.5rem;
  min-height: 2.25rem;
  background-color: ${({ theme }) => theme.colors.white};
  transition: all 0.2s ease-in-out 0s;

  vertical-align: middle;
  border: 1px solid rgba(36, 28, 21, 0.3);
  border-radius: 0.5rem;

  font-size: 1rem;
  line-height: 1.125rem;
  font-weight: normal;

  &:focus {
    border: 1px solid ${({ theme }) => theme.colors.petroleumBlue};
    box-shadow: inset 0 0 0 1px ${({ theme }) => theme.colors.petroleumBlue};
  }
`;
