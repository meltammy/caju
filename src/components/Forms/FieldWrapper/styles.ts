import styled from "styled-components";

export const ErrorMessage = styled.span`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 0.75rem;
`;

export const Container = styled.div`
  width: 100%;
`;

export const Label = styled.label`
  width: 100%;
  font-weight: 600;
`;
