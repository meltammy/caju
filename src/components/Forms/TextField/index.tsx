import { InputHTMLAttributes } from "react";
import { FieldValues, Path, UseFormRegister } from "react-hook-form";
import styled from "styled-components";

export const Input = styled.input`
  padding: 0 8px;
  vertical-align: middle;
  border-radius: 2px;
  width: 100%;
  min-height: 36px;
  background-color: ${({ theme }) => theme.colors.white};
  border: 1px solid rgba(36, 28, 21, 0.3);
  transition: all 0.2s ease-in-out 0s;
  font-size: 16px;
  line-height: 18px;
  font-weight: normal;
  border-radius: 8px;
  box-sizing: border-box;

  &:focus {
    outline: none;
    border: 1px solid ${({ theme }) => theme.colors.petroleumBlue};
    box-shadow: inset 0 0 0 1px ${({ theme }) => theme.colors.petroleumBlue};
  }
`;

type UseFormProps<FormData extends FieldValues> =
  | {
      name: Path<FormData>;
      register: UseFormRegister<FormData>;
    }
  | {
      name: never;
      register: never;
    };

export type TextFieldProps<FormData extends FieldValues> = {
  label?: string;
  error?: string;
  id: string;
} & InputHTMLAttributes<any> &
  UseFormProps<FormData>;

function TextField<FormData extends FieldValues>({
  register,
  name,
  error,
  ...props
}: TextFieldProps<FormData>) {
  const registerProps = register ? register(name) : undefined;

  const helpId = `${props.id}-help`;

  return (
    <div>
      <label htmlFor={props.id}>{props.label}</label>
      <Input
        {...props}
        {...registerProps}
        aria-invalid={!!error}
        aria-describedby={helpId}
      />
      <span role="alert" id={helpId} style={{ fontSize: 12, color: "red" }}>
        {error}
      </span>
    </div>
  );
}

export default TextField;
