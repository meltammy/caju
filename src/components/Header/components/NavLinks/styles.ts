import styled from "styled-components";

export const NavItems = styled.a`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 1.5rem;
  font-weight: 600;
`;
