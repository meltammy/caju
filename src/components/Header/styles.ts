import styled from "styled-components";

export const Container = styled.header`
  background: ${({ theme }) => theme.colors.white};
  width: 100%;
  box-shadow: 4px 4px 16px 3px #0000001f;
  height: 4rem;
  position: fixed;
  top: 0;
  display: flex;
  align-items: center;
  padding: 0px 1.5rem;
  gap: 2rem;
  z-index: 1;
`;

export const Title = styled.h1`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 1.5rem;
  font-weight: 600;
`;
