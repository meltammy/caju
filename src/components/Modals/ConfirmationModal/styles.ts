import styled from "styled-components";

export const ModalContainer = styled.div`
  min-width: 32rem;
  max-width: 95vw;
  z-index: 999;
  padding: 1.5rem;
  background: white;
  border-radius: 0.5rem;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);

  @media screen and (max-width: ${({ theme }) => theme.breakpoints.desktop}) {
    min-width: 50vw;
    margin: 1rem;
  }
`;

export const Title = styled.h2`
  margin: 0;
`;

export const Description = styled.p`
  margin: 10px 0 1.5rem;
`;

export const ButtonContainer = styled.div`
  display: flex;
  gap: 1rem;

  @media screen and (max-width: ${({ theme }) => theme.breakpoints.desktop}) {
    flex-direction: column-reverse;
  }
`;
