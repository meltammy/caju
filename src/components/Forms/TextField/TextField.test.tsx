import "@testing-library/jest-dom";

import { fireEvent, render } from "@testing-library/react";
import { useForm, FormProvider } from "react-hook-form";
import TextField from ".";
import { ThemeProvider } from "styled-components";
import { theme } from "~/../styles/theme";

function Component(props?: Partial<React.ComponentProps<typeof TextField>>) {
  const methods = useForm();

  const defaultProps = {
    id: "test-field",
    name: "testField",
    label: "Test Label",
    ...props,
  };

  return (
    <ThemeProvider theme={theme}>
      <FormProvider {...methods}>
        <TextField
          {...defaultProps}
          register={methods.register}
          name="testField"
        />
      </FormProvider>
    </ThemeProvider>
  );
}

const renderTextField = (
  props: Partial<React.ComponentProps<typeof TextField>> = {}
) => {
  return render(<Component {...props} />);
};

describe("TextField component", () => {
  it("should render label and input", () => {
    const { getByLabelText } = renderTextField();

    expect(getByLabelText(/test label/i)).toBeInTheDocument();

    const input = getByLabelText(/test label/i);
    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute("id", "test-field");
  });

  it("should render error message when provided", () => {
    const { getByText } = renderTextField({ error: "This field is required" });

    const errorMessage = getByText(/this field is required/i);
    expect(errorMessage).toBeInTheDocument();
  });

  it("should does not render error message when not provided", () => {
    const { queryByText } = renderTextField();

    const errorMessage = queryByText(/this field is required/i);
    expect(errorMessage).not.toBeInTheDocument();
  });

  it("should updates input value when typed", () => {
    const { getByLabelText } = renderTextField();
    const input = getByLabelText(/test label/i);
    const value = "New Value";
    fireEvent.change(input, { target: { value } });

    expect(input).toHaveValue(value);
  });

  it("should match snapshot", () => {
    const { asFragment } = renderTextField();
    expect(asFragment()).toMatchSnapshot();
  });
});
