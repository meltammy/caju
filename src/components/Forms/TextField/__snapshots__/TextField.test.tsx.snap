// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-egkSDF dXFjag"
  >
    <label
      aria-labelledby="test-field"
      class="sc-fAUdSK eyaFLV"
      for="test-field"
    >
      Test Label
    </label>
    <input
      aria-describedby="test-field-help"
      aria-invalid="false"
      class="sc-blHHSb bNsITS"
      id="test-field"
      name="testField"
    />
  </div>
</DocumentFragment>
`;
